snapcraft/ua_manager.py

```diff
@@ -101,8 +128,30 @@ def _is_attached() -> bool:
     return _status()["attached"]
 
 
+def _is_service_enabled(service_name: str, service_data: List[Dict[str, str]]) -> bool:
+    """Check if a service is enabled.
+
+    :param service_name: name of service to check
+    :param service_data: list of dictionaries containing data for each service
+
+    :return: True if the service is enabled. False is the service is disabled or not
+    listed in the list of service data.
+    """
+    service = next(
+        (service for service in service_data if service.get("name") == service_name),
+        None,
+    )
+
+    if service:
+        # possible statuses are 'enabled', 'disabled', and 'n/a'
+        return service.get("status") == "enabled"
```

Comment:
This logic feels a little too complext. What about something like the following?
```
    for service in service_data:
        if service.get("name") == service_name):
            # possible statuses are 'enabled', 'disabled', and 'n/a'
            return service.get("status") == "enabled"

    emit.debug(f"Could not find service {service_name!r}.")
    return False
```

Reply:
Good idea. Simplified.